src/main/java/uk/nhs/hee/tis/revalidation/core/service/AdminService.java

```diff
@@ -60,9 +62,34 @@
 
     // TODO: A limited number of users can be returned before pagination occurs, handle pagination.
     ListUsersInGroupResult listUsersResult = identityProvider.listUsersInGroup(request);
+    List<UserType> userTypeList = listUsersResult.getUsers();
 
-    return listUsersResult.getUsers().stream()
-        .map(mapper::toDto)
-        .collect(Collectors.toList());
+    return buildAdminDtoList(userTypeList);
+
+  }
+
+  private List<AdminDto> buildAdminDtoList(List<UserType> userTypeList) {
+    List<AdminDto> adminDtoList = new ArrayList<>();
+    for (UserType userType : userTypeList) {
```

Comment:
Please use streaming instead of this loop.

Reply:
@uzairbaig I have added streaming